### DISCUSSION TITLE
"int_steps" parameter affecting model prediction size

### DISCUSSION THREAD

**Discussion by Kanyq** [voxelmorph/voxelmorph#299] [Q&A] [answered]
Hello,

I was experimenting with the "int_size" parameter of the VxmDense class and to build my model
(vxm_model = vxm.networks.VxmDense(vol_shape, nb_features, int_steps=0)) and when I set the "int_size" to any value other than 0 the "second" predicted array of the model gets reduced in size by half.

Original dimension of training and test images: 256,256,64

val_pred = vxm_model.predict(val_input)
moved_pred = val_pred[0]
pred_warp = val_pred[1]

size of "pred_warp" : 128,128,32

I'm not sure how or why the "int_steps" parameter is affecting the training so any help in understanding that would be very appreciated.

**Comment by adalca** [comment, marked as answer]
the second output from that model isn't the warp itself necessarily:
if int_steps > 0, then the second output is the velocity field (which we output so we can penalize for smoothness). By default, it's resized by `int_downsize=2` to make integration faster and more memory efficient. To get the deformation field itself, you can use `vxm_model.get_registration_model()`. 

if int_steps = 0, then we just output the deformation field.
